graphql/util.go

```diff
@@ -909,6 +918,54 @@ func getHostRequestOptions(ctx context.Context, usr *user.DBUser, spawnHostInput
 	return options, nil
 }
 
+// GetProjectAndModulesForTask returns the project ref and modules for a task
+func getProjectAndModulesForTask(ctx context.Context, settings *evergreen.Settings, t *task.Task) (*model.ProjectRef, []host.ProjectModule, error) {
+	if t == nil {
+		return nil, nil, errors.New("task is nil")
+	}
+	projectRef, err := model.FindBranchProjectRef(t.Project)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "finding project ref")
+	}
+
+	v, err := model.VersionFindOne(model.VersionById(t.Version))
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "finding version")
+	}
+	pp, err := model.ParserProjectFindOneByID(ctx, settings, v.ProjectStorageMethod, v.Id)
+	if err != nil || pp == nil {
+		return nil, nil, errors.Wrap(err, "finding project ref")
+	}
+	// Then fetch each of the modules
+	project, err := model.TranslateProject(pp)
```

Comment:
Do we need to re-translate the entire project just to get the module info? This could be pretty expensive, so it would be good to avoid if possible?

Rather than re-translate the whole project, is it possible to just query the version manifest to figure out what modules are relevant to the task?